lib/elixir/lib/task.ex

```diff
@@ -515,13 +547,16 @@ defmodule Task do
 
   """
   @spec await(t, timeout) :: term
-  def await(task, timeout \\ 5000)
+  def await(%Task{} = task, timeout \\ 5000)
```

Comment:
Two comments here:

1. I think we could use `defguardp` here to define a `is_timeout`
2. I'm not sure if this kind of defensive programming is needed here (I like it for my personal projects, but I'm not sure if that's also the preference in the Elixir codebase). If you pass an invalid value in `timeout`, it will raise `** (ErlangError) Erlang error: :timeout_value` which I think is self-explanatory (or maybe we don't want to raise `ErlangError`?).

Reply:
:+1: for the `is_timeout` guard. I think having a guard is good, so let's keep it.

Reply:
done